powertools-sqs/src/test/java/software/amazon/lambda/powertools/sqs/SqsUtilsFifoBatchProcessorTest.java

```diff
@@ -0,0 +1,122 @@
+package software.amazon.lambda.powertools.sqs;
+
+import com.amazonaws.services.lambda.runtime.events.SQSEvent;
+import com.fasterxml.jackson.databind.ObjectMapper;
+import org.junit.jupiter.api.AfterEach;
+import org.junit.jupiter.api.BeforeEach;
+import org.junit.jupiter.api.Test;
+import org.mockito.*;
+import org.mockito.quality.Strictness;
+import software.amazon.awssdk.services.sqs.SqsClient;
+import software.amazon.awssdk.services.sqs.model.DeleteMessageBatchRequest;
+import software.amazon.awssdk.services.sqs.model.DeleteMessageBatchRequestEntry;
+import software.amazon.awssdk.services.sqs.model.DeleteMessageBatchResponse;
+
+import java.io.IOException;
+import java.util.List;
+import java.util.concurrent.atomic.AtomicInteger;
+import java.util.stream.Collectors;
+
+import static org.assertj.core.api.Assertions.assertThat;
+import static org.assertj.core.api.Assertions.assertThatExceptionOfType;
+import static org.mockito.Mockito.verifyNoMoreInteractions;
+import static org.mockito.MockitoAnnotations.openMocks;
+import static software.amazon.lambda.powertools.sqs.SqsUtils.batchProcessor;
+import static software.amazon.lambda.powertools.sqs.SqsUtils.overrideSqsClient;
+
+public class SqsUtilsFifoBatchProcessorTest {
+
+    private static final ObjectMapper MAPPER = new ObjectMapper();
+    private static SQSEvent sqsBatchEvent;
+    private MockitoSession session;
+
+    @Mock
+    private SqsClient sqsClient;
+
+    @Captor
+    private ArgumentCaptor<DeleteMessageBatchRequest> deleteMessageBatchCaptor;
+
+    public SqsUtilsFifoBatchProcessorTest() throws IOException {
+        sqsBatchEvent  =  MAPPER.readValue(SqsUtilsFifoBatchProcessorTest.class.getResource("/SqsFifoBatchEvent.json"), SQSEvent.class);
```

Comment:
why don't you use the https://github.com/aws/aws-lambda-java-libs/tree/main/aws-lambda-java-tests ?

Reply:
Done! I load it once in the constructor (i've only got one event to use for now).